app/controllers/api/v1/private/publishers_controller.rb

```diff
@@ -0,0 +1,78 @@
+class Api::V1::Private::PublishersController < Api::V1::Private::BaseController
+  class InvalidNote < StandardError; end
+  class InvalidAdmin < StandardError; end
+  def show
+    publisher = Publisher.find(params[:id])
+    current_status_update = publisher.last_status_update
+    data = {
+      id: publisher.id,
+      owner_identifier: publisher.owner_identifier,
+      email: publisher.email,
+      name: publisher.name,
+      created_at: publisher.created_at,
+      current_status: {
+        status: current_status_update.status,
+        created_at: current_status_update.created_at,
+      },
+    }
+    render(status: 200, json: data) and return
+
+  rescue ActiveRecord::RecordNotFound
+    error_response = {
+      errors: [{
+        status: "404",
+        title: "Not Found",
+        detail: "Publisher with id #{params[:publisher_id]} not found",
+      },],
+    }
+
+    render(status: 404, json: error_response) and return
+  end
+
+  def publisher_status_updates
+    publisher = Publisher.find(params[:publisher_id])
+    admin = Publisher.find_by_email(params[:admin])
+    status = params[:status]
+    note = params[:note]
+
+    raise InvalidNote if note.blank?
+    raise InvalidAdmin if admin.blank?
+
+    PublisherStatusUpdate.create!(publisher: publisher, status: status)
+    PublisherNote.create!(note: note, publisher: publisher, created_by: admin)
+
+    render(status: 200, json: { success: "true" }) and return
+
+  rescue ActiveRecord::RecordInvalid
```

Comment:
I think it would be better to have the models check the validation instead. Something like 

```ruby 
status = PublisherStatusUpdate.new(publisher: publisher, status: status)
note =  PublisherNote.new(note: note, publisher: publisher, created_by: admin)
if status.valid? && note.valid?
  status.save
  note.save
  render json: { success: "true" }
else
  render json: { status_errors: status.errors.messages, note_errors: note.errors.messages }
end
```

Then Rails will take care of all the error messaging and stuff for us

Reply:
I felt the same way and built it that way first, but I like giving the external service more detailed messages